### PR TITLE
Register login

### PR DESCRIPTION
**Pull request by paolovic**
Not perfect, will be improved, but now you've got the endpoints for registration and login.
Tests are also missing yet.
Authentication is missing.
Password is stored in clear text in the db.
@luis-tm @mkorob

### CODE REVIEW

File: src/main/java/ch/uzh/ifi/hase/soprafs22/rest/dto/get_dto/UserRegistrationGetDTO.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package ch.uzh.ifi.hase.soprafs22.rest.dto.get_dto;
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+
+import java.util.Date;
+
+public class UserRegistrationGetDTO {
+    private long id;
+    private String username;
+    @JsonFormat(pattern="yyyy-MM-dd")
```
Comment (apple00juice) [inline src/main/java/ch/uzh/ifi/hase/soprafs22/rest/dto/get_dto/UserRegistrationGetDTO.java:10]: ohhh, good to know, that annoyed me really during M1!

**Review by apple00juice — APPROVED** [review]
Rest looks perfect! 👍

File: src/main/java/ch/uzh/ifi/hase/soprafs22/user/RegisteredUser.java [hunk 2]
```diff
@@ -24,6 +27,14 @@ public class RegisteredUser implements Serializable {
     @Column(nullable = false)
     private String password;
 
+    @Column(name = "logged_in", nullable = false)
+    private boolean loggedIn;
```
[the remark this answers is not in the document]
Comment (paolovic) [reply src/main/java/ch/uzh/ifi/hase/soprafs22/user/RegisteredUser.java:31]: i'm not sure.
main reason for implementing it was to do some action with / to test the login endpoint.
at the moment i wouldn't have done anything meaningful when logging in.

**Review by mkorob — APPROVED** [review]

**Review by luis-tm — APPROVED** [review]

File: build.gradle [hunk 3]
```diff
@@ -52,6 +53,10 @@ dependencies {
     implementation 'org.springframework.boot:spring-boot-starter-websocket'
 
     implementation 'org.springframework.boot:spring-boot-starter-validation'
+    implementation 'org.springframework.boot:spring-boot-starter-security:2.7.0'
```
Comment (hilalcomak) [inline build.gradle:56]: When I removed this security module all tests pass again. Looking around
https://stackoverflow.com/questions/46745923/why-cant-i-import-withmockuser-to-my-test suggests importing security-test
https://docs.spring.io/spring-security/site/docs/4.2.x/reference/html/test-method.html
https://www.baeldung.com/spring-security-integration-tests

I think we need these 2, but couldn't get them to work
testImplementation('org.springframework.boot:spring-boot-starter-test')
testImplementation('org.springframework.security:spring-security-test')

Comment (paolovic) [reply build.gradle:56]: Thank you for your work! 👍🏻

**Pull request merged**